__tests__/app.test.js

```diff
@@ -31,9 +33,377 @@ describe("NCNews-Server Unit Tests", () => {
+            });
+
+        });
+
+
+    });         
+
+    // Task 8 - PATCH /api/articles/:article_id
+    describe(`PATCH /api/articles/:article_id`, () => {
+
+        const defaultVotesObject = { inc_votes : 1 };
+
+        test(`[ 201 ] Responds with the updated article with correctly adjusted vote count, increasing vote count`, () => {
+            const votesObject = { inc_votes : 2 };
+
+            return request(app).patch('/api/articles/1').send(votesObject).expect(201).then((response) => {
+                
+                expect(response.body.article["article_id"]).toEqual(1);
+                expect(response.body.article.votes).toEqual(102);
+            });
+
+        });
+
+        test(`[ 201 ] Responds with the updated article with correctly adjusted vote count, decreasing vote count`, () => {
+            const votesObject = { inc_votes : -5 };
+
+            return request(app).patch('/api/articles/1').send(votesObject).expect(201).then((response) => {
+
+                expect(response.body.article["article_id"]).toEqual(1);
+                expect(response.body.article.votes).toEqual(95);
+            });
+
+        });
+
+        test(`[ 400 ] Responds with an error when passed invalid URL parameters`, () => {
+
+            return request(app).patch('/api/articles/fddfggdrtd').send(defaultVotesObject).expect(400).then((response) => {
+                expect(response.body).toEqual({ error: "Error 400! BAD REQUEST" });
+            });
+
+        });
+
+        test(`[ 400 ] Responds with an error when passed an object without the required keys`, () => {
+
+            const votesObject = {
+                wat: "k",
+                k: "k"
+            };
+
+            return request(app).post('/api/articles/1/comments').send(votesObject).expect(400).then((response) => {
+                expect(response.body).toEqual({ error: "Error 400! BAD REQUEST" });
+            });
+
+        });
+
+        test(`[ 400 ] Responds with an error when passed an invalid vote of the wrong type`, () => {
+
+            const votesObject = {
+                inc_votes: `dksfsdjlfds`
+            };
+
+            return request(app).post('/api/articles/1/comments').send(votesObject).expect(400).then((response) => {
+                expect(response.body).toEqual({ error: "Error 400! BAD REQUEST" });
+            });
+
+
+        });
+
+        test(`[ 400 ] Responds with an error when passed an SQL injection in the votes property`, () => {
+
+            const votesObject = {
+                inc_votes: `1; DROP TABLE articles;`
+            };
+
+            return request(app).post('/api/articles/1/comments').send(votesObject).expect(400).then((response) => {
+                expect(response.body).toEqual({ error: "Error 400! BAD REQUEST" });
+            });
+
+
+        });
+
+        
+        test(`[ 400 ] Responds with an error when passed an SQL injection in URL parameters, test 2`, () => {
```

Comment:
please dont leave this test in, this is not testing for SQL injection, this is the same test as passing an incorrect article_id data type. We dont know what sql injection actually looks like so we can't guarantee that this is an effective test

Reply:
test removed. thank you for explaining!